packages/rum-vue/src/route-hooks.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * MIT License
+ *
+ * Copyright (c) 2017-present, Elasticsearch BV
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+export function routeHooks(router, apm) {
+  let transaction
+  router.beforeResolve((to, from, next) => {
```

Comment:
I think we should use `beforeEach` here instead, `beforeResolve` is called after the route is "confirmed", which includes checking other `beforeEach` hooks and since each of these hooks could resolve asynchronously the whole route change could take longer because of these.

Reply:
We should not do `beforeEach` since the user might cancel the current navigation and can install route guards before that. Like checking for authentication and so on. `beforeResolve` accounts for those cases as well and starts once navigation is confirmed with all checks. Its the safest.

Reply:
Vue router calls `beforeEach` and `beforeResolve` before our history.pushstate works so we might miss some fetches that happen in components `beforeEach`. I am moving our logic to `beforeEach` which should fix the problem. In the worst case, if the component beforeEach logic takes more than 5 seconds which is our reuse interval. we might end up creating new transaction to capture the route-change. Its not a huge deal though, since the users would call `next(nextRoute)` in an async way on each hook, so we should be good here. WDYT @jahtalab ?